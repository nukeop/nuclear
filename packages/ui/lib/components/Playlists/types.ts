export enum PlaylistsColumn {
    Thumbnail = 'thumbnail',
    Title = 'title',
    Duration = 'duration',
    Tracks = 'tracks',
    LastModified = 'lastModified',
    Sync = 'sync'
}
