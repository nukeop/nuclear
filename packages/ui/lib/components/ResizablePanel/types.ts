type ResizablePanelClasses = {
    root: string;
    content: string;
}

export type ResizablePanelProps = {
  classes?: Partial<ResizablePanelClasses>;
  width: number;
  onSetWidth?: (width: number) => void;
  isCollapsed?: boolean;
  collapsedWidth: number;
  right?: boolean;
};
