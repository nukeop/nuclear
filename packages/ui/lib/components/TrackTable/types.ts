import { Track } from '../../types';
import { TrackPopupStrings } from '../TrackPopup';

export enum TrackTableColumn {
  Delete = 'delete',
  Position = 'position',
  Thumbnail = 'thumbnail',
  Favorite = 'favorite',
  Title = 'title',
  Artist = 'artist',
  Album = 'album',
  Duration = 'duration',
  Selection = 'selection'
}

export type TrackTableExtraProps = {
  onPlay?: (track: Track) => void;
  onPlayNext?: (track: Track) => void;
  onPlayAll?: (tracks: Track[]) => void;
  onAddToQueue?: (track: Track) => void;
  onAddToFavorites?: (track: Track) => void;
  onRemoveFromFavorites?: (track: Track) => void;
  onAddToPlaylist?: (track: Track, { name: string }) => void;
  onAddToDownloads?: (track: Track) => void;
  onDelete?: (track: Track, idx: number) => void;
  playlists?: Array<{
    name: string;
  }>;
  popupActionStrings?: TrackPopupStrings
}

export type TrackTableStrings = {
  addSelectedTracksToQueue: string;
  playSelectedTracksNow: string;
  addSelectedTracksToFavorites: string;
  addSelectedTracksToDownloads: string;
  tracksSelectedLabelSingular: string;
  tracksSelectedLabelPlural: string;
}

export type TrackTableHeaders = {
  positionHeader: React.ReactNode;
  thumbnailHeader: React.ReactNode;
  artistHeader: string;
  titleHeader: string;
  albumHeader: string;
  durationHeader: string;
}

export type TrackTableSettings = {
  displayDeleteButton?: boolean;
  displayPosition?: boolean;
  displayThumbnail?: boolean;
  displayFavorite?: boolean;
  displayArtist?: boolean;
  displayAlbum?: boolean;
  displayDuration?: boolean;
  selectable?: boolean;
};


