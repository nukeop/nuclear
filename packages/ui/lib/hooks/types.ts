import { useForm } from './useForm';

export type FormValue = string | number | boolean;
export type FieldsPropsType = {
  [k: string]: {
    name: string;
    label: string;
    placeholder?: string;
    value: FormValue;
    error?: string;
    onChange: (value: FormValue, event?: React.ChangeEvent) => void;
    onBlur: React.FocusEventHandler;
  }
};

export type FormProps = Partial<ReturnType<typeof useForm>>;

export type InitialFormField = {
  name: string;
  label?: string;
  placeholder?: string;
  initialValue?: FormValue;
}

export type InitialFormFields = {
  [k: string]: InitialFormField;
}
