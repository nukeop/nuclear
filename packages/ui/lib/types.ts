export type Track = {
  local?: boolean;
  album?: string;
  artist?: { name: string } | string;
  duration?: number | string;
  position?: number | string;
  playcount?: number | string;
  title?: string;
  name?: string;
  thumbnail?: string;
};
