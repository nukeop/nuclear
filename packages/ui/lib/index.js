import '../resources/Lato.css';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Card } from './components/Card';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as SmoothImage } from './components/SmoothImage';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as PopupButton } from './components/PopupButton';
export { default as QueueItem } from './components/QueueItem';
export { default as QueuePopup } from './components/QueuePopup';
export { default as Range } from './components/Range';
export { default as Seekbar } from './components/Seekbar';
export { default as ToastContainer } from './components/ToastContainer';
export { default as TrackRow } from './components/TrackRow';
export { default as UserPluginsItem } from './components/UserPluginsItem';

export * from './utils';
