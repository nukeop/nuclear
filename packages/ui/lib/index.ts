import '../resources/Lato.scss';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as ArtistGrid } from './components/ArtistGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Button } from './components/Button';
export { default as Card } from './components/Card';
export { default as Checkbox } from './components/Checkbox';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as Dropdown } from './components/Dropdown';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as MiniPlayer } from './components/MiniPlayer';
export { default as PlayerBar } from './components/PlayerBar';
export { default as PlayerButton } from './components/PlayerButton';
export { default as PopupButton } from './components/PopupButton';
export { default as PopupDropdown } from './components/PopupDropdown';
export { default as QueueItem } from './components/QueueItem';
export { default as Range } from './components/Range';
export { default as SearchBox } from './components/SearchBox';
export { default as Seekbar } from './components/Seekbar';
export { default as SmoothImage } from './components/SmoothImage';
export { default as StreamInfo } from './components/StreamInfo';
export { default as ToastContainer } from './components/ToastContainer';
export { default as Tooltip } from './components/Tooltip';
export { default as TrackPopup } from './components/TrackPopup';
export { default as TrackRow } from './components/TrackRow';
export { default as TrackTable } from './components/TrackTable';
export { default as UserPluginsItem } from './components/UserPluginsItem';
export { default as Visualizer } from './components/Visualizer';
export { default as VisualizerOverlay } from './components/VisualizerOverlay';

export * from './utils';
