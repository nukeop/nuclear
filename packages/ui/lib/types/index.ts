export type Track = {
  uuid?: string;
  loading?: boolean;
  error?: boolean | { 
    message: string; 
    details: string 
  };
  local?: boolean;
  artist: { name: string } | string;
  name?: string;
  title?: string;
  album?: string;
  duration?: number | string;
  position?: number | string;
  playcount?: number | string;
  thumbnail?: string;
  image?: { '#text'?: string }[];
  streams?: TrackStream[];
};

export type Album = {
  artist?: { name: string };
  title: string;
  tracks?: Track[];
  image?: { '#text'?: string }[];
};

export type TrackStream = {
  source: string;
  id: string;
  duration?: number;
  title?: string;
  thumbnail?: string;
  stream?: string;
};

export type SearchProviderOption = {
  key: string;
  text: string;
  value: string;
}

export type TrackItem = {
  artist: string;
  name: string;
  thumbnail?: string;
  local?: boolean;
  uuid?: string;
  streams?: TrackStream[];
}

export type Download = {
  status: DownloadStatus;
  completion: number;
  track: TrackItem;
}
export enum DownloadStatus {
  WAITING = 'Waiting',
  STARTED = 'Started',
  PAUSED = 'Paused',
  FINISHED = 'Finished',
  ERROR = 'Error'
}

export type Notification = {
  id?:string
  title?: string;
  details?: string;
  error?: boolean;
  warning?: boolean;
  info?: boolean;
  success?: boolean;
  onClick?:() => void
  icon?:  Node | React.ReactElement<{
    src: string;
  }>
}
