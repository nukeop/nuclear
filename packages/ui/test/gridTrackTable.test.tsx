import { GridTrackTable } from '../lib';
import { Track } from '../lib/types';
import { makeSnapshotTest } from './helpers';

jest.mock(
  'react-virtualized-auto-sizer',
  () =>
    ({ children }) =>
      children({ height: 600, width: 600 })
);

makeSnapshotTest(GridTrackTable, {
  tracks: [],
  positionHeader: 'Position',
  thumbnailHeader: 'Thumbnail',
  artistHeader: 'Artist',
  albumHeader: 'Album',
  titleHeader: 'Title',
  durationHeader: 'Length'
}, '(Snapshot) Grid track table - empty');

makeSnapshotTest(GridTrackTable, {
  tracks: [
    {
      position: 1, 
      thumbnail: 'https://i.imgur.com/4euOws2.jpg', 
      artists: ['Test Artist'],
      title: 'Test Title',
      album: 'Test Album',
      duration: '1:00'
    }, {
      position: 2, 
      thumbnail: 'https://i.imgur.com/4euOws2.jpg', 
      artists: ['Test Artist 2', 'Test Artist 4'],
      name: 'Test Title 2',
      album: 'Test Album',
      duration: '1:00'
    } as Track,
    {
      position: 3, 
      thumbnail: 'https://i.imgur.com/4euOws2.jpg', 
      artists: ['Test Artist 3'],
      name: 'Test Title 3',
      album: 'Test Album',
      duration: '1:00'
    } as Track
  ],
  positionHeader: 'Position',
  thumbnailHeader: 'Thumbnail',
  artistHeader: 'Artist',
  albumHeader: 'Album',
  titleHeader: 'Title',
  durationHeader: 'Length'
}, '(Snapshot) Grid track table - example data with all rows');
