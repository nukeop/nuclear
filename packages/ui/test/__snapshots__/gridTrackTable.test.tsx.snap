// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Grid track table - empty should render correctly 1`] = `
<DocumentFragment>
  <div
    class="track_table_wrapper"
  >
    <div
      class="grid_track_table"
      role="table"
    >
      <div
        class="track_table_head"
      >
        <div
          class="track_table_header_row"
          role="row"
          style="grid-template-columns: 3em 4em 3em 3em minmax(8em, 1fr) 1fr 1fr 6em 6em auto;"
        >
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
          >
             
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header centered"
              data-testid="position-header"
            >
              Position
              <i
                aria-hidden="true"
                class="sort content ascending icon text_header_icon"
              />
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Thumbnail
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
          >
             
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Title
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Artist
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Album
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Length
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
          >
            <div
              class="selection_header"
            >
              <div
                class="ui fitted checkbox nuclear checkbox"
                style="cursor: pointer;"
              >
                <input
                  class="hidden"
                  readonly=""
                  tabindex="0"
                  title="Toggle All Rows Selected"
                  type="checkbox"
                  value=""
                />
                <label />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="track_table_body"
        data-rbd-droppable-context-id="0"
        data-rbd-droppable-id="track_table"
        data-testid="track-table-body"
        role="rowgroup"
      >
        <div
          class="track_table_virtualized_list"
          style="position: relative; height: 600px; width: 600px; overflow: auto; will-change: transform; direction: ltr;"
        >
          <div
            style="height: 0px; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Grid track table - example data with all rows should render correctly 1`] = `
<DocumentFragment>
  <div
    class="track_table_wrapper"
  >
    <div
      class="grid_track_table"
      role="table"
    >
      <div
        class="track_table_head"
      >
        <div
          class="track_table_header_row"
          role="row"
          style="grid-template-columns: 3em 4em 3em 3em minmax(8em, 1fr) 1fr 1fr 6em 6em auto;"
        >
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
          >
             
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header centered"
              data-testid="position-header"
            >
              Position
              <i
                aria-hidden="true"
                class="sort content ascending icon text_header_icon"
              />
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Thumbnail
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
          >
             
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Title
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Artist
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Album
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="text_header"
            >
              Length
            </div>
          </div>
          <div
            class="track_table_header_cell"
            colspan="1"
            role="columnheader"
          >
            <div
              class="selection_header"
            >
              <div
                class="ui fitted checkbox nuclear checkbox"
                style="cursor: pointer;"
              >
                <input
                  class="hidden"
                  readonly=""
                  tabindex="0"
                  title="Toggle All Rows Selected"
                  type="checkbox"
                  value=""
                />
                <label />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="track_table_body"
        data-rbd-droppable-context-id="1"
        data-rbd-droppable-id="track_table"
        data-testid="track-table-body"
        role="rowgroup"
      >
        <div
          class="track_table_virtualized_list"
          style="position: relative; height: 600px; width: 600px; overflow: auto; will-change: transform; direction: ltr;"
        >
          <div
            style="height: 126px; width: 100%;"
          >
            <div
              class="grid_track_table_row"
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="0"
              data-testid="track-table-row"
              role="row"
              style="position: absolute; left: 0px; top: 0px; height: 42px; width: 100%; grid-template-columns: 3em 4em 3em 3em minmax(8em, 1fr) 1fr 1fr 6em 6em;"
              tabindex="0"
            >
              <div
                class="delete_cell"
                data-testid="delete-cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </div>
              <div
                class="grid_track_table_cell position_cell"
                data-testid="position-cell"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button play_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  1
                </span>
              </div>
              <div
                class="grid_track_table_cell thumbnail_cell"
                role="cell"
              />
              <div
                class="grid_track_table_cell favorite_cell"
                data-testid="favorite-cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="favorite-button"
                >
                  <i
                    aria-hidden="true"
                    class="heart outline icon"
                  />
                </button>
              </div>
              <div
                class="grid_track_table_cell text_cell title_cell"
                data-testid="title-cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    Test Title
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                Test Artist
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                Test Album
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                1:00
              </div>
              <div
                class="selection_cell"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="-1"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </div>
            </div>
            <div
              class="grid_track_table_row"
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="1"
              data-testid="track-table-row"
              role="row"
              style="position: absolute; left: 0px; top: 42px; height: 42px; width: 100%; grid-template-columns: 3em 4em 3em 3em minmax(8em, 1fr) 1fr 1fr 6em 6em;"
              tabindex="1"
            >
              <div
                class="delete_cell"
                data-testid="delete-cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </div>
              <div
                class="grid_track_table_cell position_cell"
                data-testid="position-cell"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button play_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  2
                </span>
              </div>
              <div
                class="grid_track_table_cell thumbnail_cell"
                role="cell"
              />
              <div
                class="grid_track_table_cell favorite_cell"
                data-testid="favorite-cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="favorite-button"
                >
                  <i
                    aria-hidden="true"
                    class="heart outline icon"
                  />
                </button>
              </div>
              <div
                class="grid_track_table_cell text_cell title_cell"
                data-testid="title-cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    Test Title 2
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                Test Artist 2
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                Test Album
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                1:00
              </div>
              <div
                class="selection_cell"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="-1"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </div>
            </div>
            <div
              class="grid_track_table_row"
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="2"
              data-testid="track-table-row"
              role="row"
              style="position: absolute; left: 0px; top: 84px; height: 42px; width: 100%; grid-template-columns: 3em 4em 3em 3em minmax(8em, 1fr) 1fr 1fr 6em 6em;"
              tabindex="2"
            >
              <div
                class="delete_cell"
                data-testid="delete-cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="delete-button"
                >
                  <i
                    aria-hidden="true"
                    class="times icon"
                  />
                </button>
              </div>
              <div
                class="grid_track_table_cell position_cell"
                data-testid="position-cell"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button play_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  3
                </span>
              </div>
              <div
                class="grid_track_table_cell thumbnail_cell"
                role="cell"
              />
              <div
                class="grid_track_table_cell favorite_cell"
                data-testid="favorite-cell"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                  data-testid="favorite-button"
                >
                  <i
                    aria-hidden="true"
                    class="heart outline icon"
                  />
                </button>
              </div>
              <div
                class="grid_track_table_cell text_cell title_cell"
                data-testid="title-cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    Test Title 3
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                Test Artist 3
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                Test Album
              </div>
              <div
                class="grid_track_table_cell text_cell"
                role="cell"
              >
                1:00
              </div>
              <div
                class="selection_cell"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="-1"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
